source/base/bounding_box.cc

```diff
@@ -42,6 +42,83 @@ void BoundingBox<spacedim,Number>::merge_with(const BoundingBox<spacedim,Number>
 }
 
 template <int spacedim, typename Number>
+NeighbourType BoundingBox < spacedim,Number >::is_neighbour(const BoundingBox<spacedim,Number> &other_bbox) const
+{
+  if (spacedim == 1)
+    {
+      // In dimension 1 if the two bounding box are neighbours
+      // we can merge them
+      if ( this->point_inside(other_bbox.boundary_points.first)
+           || this->point_inside(other_bbox.boundary_points.second) )
+        return mergeable_neighbours;
+      return not_neighbours;
+    }
+  else
+    {
+      std::vector< Point<spacedim,Number> > bbox1;
+      bbox1.push_back(this->get_boundary_points().first);
+      bbox1.push_back(this->get_boundary_points().second);
+      std::vector< Point<spacedim,Number> > bbox2;
+      bbox2.push_back(other_bbox.get_boundary_points().first);
+      bbox2.push_back(other_bbox.get_boundary_points().second);
+
+      // Step 1: testing if the boxes are close enough to intersect
+      for ( unsigned int d=0 ; d<spacedim; ++d )
+        if ( bbox1[0][d] - std::numeric_limits<double>::epsilon() > bbox2[1][d]
+             || bbox2[0][d] - std::numeric_limits<double>::epsilon() > bbox1[1][d] )
+          return not_neighbours;
+
+      // The boxes intersect: we need to understand now how they intersect.
+      // We begin by computing the intersection:
+      std::vector<double> intersect_bbox_min;
+      std::vector<double> intersect_bbox_max;
+      for (unsigned int d=0; d < spacedim ; ++d)
+        {
+          intersect_bbox_min.push_back(std::max(bbox1[0][d],bbox2[0][d]));
+          intersect_bbox_max.push_back(std::min(bbox1[1][d],bbox2[1][d]));
+        }
+
+      // Finding the intersection's dimension
+
+      unsigned int intersect_dim = spacedim;
+      for (unsigned int d=0; d < spacedim; ++d)
+        if ( std::abs( intersect_bbox_min[d] - intersect_bbox_max[d] ) < std::numeric_limits<double>::epsilon() )
```

Comment:
A similar issue arises here: you want the epsilon to be *relative* to the sizes of the objects. You can achieve this by testing
```
  std::abs( intersect_bbox_min[d] - intersect_bbox_max[d] ) 
  < 
  std::numeric_limits<double>::epsilon() * (std::abs( intersect_bbox_min[d]) + std::abs(intersect_bbox_max[d] ))
```

What happens if `intersect_bbox_min[d] == intersect_bbox_max[d]`? Would it be better to use `<=` instead of `<`?